description: 'io_uring: fix race between timeout flush and removal'
references:
- https://www.openwall.com/lists/oss-security/2022/04/22/3
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-29582
- https://www.openwall.com/lists/oss-security/2022/04/22/4
- https://git.kernel.org/linus/e677edbcabee849bfdd43f1602bccbecf736a646
- https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=e677edbcabee849bfdd43f1602bccbecf736a646
- https://github.com/torvalds/linux/commit/e677edbcabee849bfdd43f1602bccbecf736a646
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.17.3
- http://www.openwall.com/lists/oss-security/2022/04/22/4
- https://ruia-ruia.github.io/2022/08/05/CVE-2022-29582-io-uring/
- https://github.com/Ruia-ruia/CVE-2022-29582-Exploit
- https://www.cve.org/CVERecord?id=CVE-2022-29582
comments:
  debian/carnil: For 5.17.y fixed as well in 5.17.3.
  masami: io_uring was introduced since 5.1. 5.4 doesn't have io_flush_timeouts().
fixed-by:
  cip/5.10: [2827328e646d0c2d3db1bfcad4b5f5016ce0d643]
  cip/5.10-rt: [2827328e646d0c2d3db1bfcad4b5f5016ce0d643]
  mainline: [e677edbcabee849bfdd43f1602bccbecf736a646]
  stable/5.10: [2827328e646d0c2d3db1bfcad4b5f5016ce0d643]
  stable/5.15: [ba7261af2b030ab2c06189be1fc77b273716839f]
  stable/5.17: [11cd7959400258beb1dc17c8680055966263f316]
  ubuntu/focal: ['version:5.4.0-9.12']
ignore:
  cip/4.19: Not affected
  cip/4.19-rt: Not affected
  cip/4.4: Not affected
  cip/4.4-rt: Not affected
  cip/4.4-st: Not affected
  stable/5.4: Not affected
